distribution/lib/Standard/Google_Api/0.0.0-dev/src/Google_Analytics_Account.enso

```diff
@@ -0,0 +1,108 @@
+    is_deleted : Boolean
+    is_deleted self = self.java_record.deleted
+
+    ## PRIVATE
+    to_text : Text
+    to_text self = "Google_Analytics_Account " + self.name + " (" + self.id + ")"
+
+    ## PRIVATE
+    to_display_text : Text
+    to_display_text self = "GA Account {" + self.name + " (" + self.id + ")}"
+
+    ## ICON data_input
+       List of all properties of the account.
+
+       Arguments:
+       - credentials: The Google credentials to use. Default is the default
+         credentials.
+       - limit: The maximum number of accounts to read. Default is 1000.
+       - include_deleted: Whether to include deleted accounts. Default is false.
+    @credentials Google_Credential.default_widget
+    @limit Rows_To_Read.default_widget
+    properties : Google_Credential -> Rows_To_Read -> Boolean -> Vector
+    properties self credential:Google_Credential=..Default (limit : Rows_To_Read = ..First_With_Warning 1000) include_deleted:Boolean=False =
+        Google_Analytics.list_properties self credential limit include_deleted
+
+## Filter for Google Analytics accounts.
+type Google_Analytics_Account_Filter
+    All_Accounts
+
+    ## Filter to a single account
+    @account (Text_Input display=..Always)
+    Account account:Text|Google_Analytics_Account
+
+    ## Filter to set of accounts
+    @accounts (Vector_Editor item_editor=Text_Input item_default="''")
+    Accounts accounts:Vector
+
+    ## PRIVATE
+    default_widget : Any -> Any -> Widget
+    default_widget self_arg cache =
+        _ = self_arg
+        default = [Option "<All>" "..All_Accounts", Option "<Account>" "..Account", Option "<Multiple Accounts>" "..Accounts"]
+        Panic.catch Any handler=_->(make_single_choice default) <|
+            credentials = cache.if_not_nothing <| cache "credentials"
+            options = Google_Analytics.list_accounts credentials=(credentials.if_nothing Google_Credential.Default) limit=(..First 10) . map p->(Option (p.name + " (" + p.id + ")") "..Account "+p.id.pretty)
+            make_single_choice default+options display=..Always
+
+   ## PRIVATE
+   as_java : Array | Nothing
+   private as_java self = case self of
+       Google_Analytics_Account_Filter.All_Accounts -> Nothing
+       Google_Analytics_Account_Filter.Account acc -> case acc of
+           _ : Google_Analytics_Account -> [acc.java_record].to_array
+           _ : Text ->
+               id = if acc.starts_with "accounts/" then acc else
+                   "accounts/" + acc
+               [AnalyticsAccount.new id "MOCK" False Date_Time.now "XXX"].to_array
+       Google_Analytics_Account_Filter.Accounts accounts ->
+           vec = accounts.map acc-> case acc of
+               _ : Google_Analytics_Account -> acc.java_record
+               _ : Text ->
+                   id = if acc.starts_with "accounts/" then acc else
+                       "accounts/" + acc
+                   AnalyticsAccount.new id "MOCK" False Date_Time.now "XXX"
```

Comment:
Duplicated logic with the `Account` branch above. Maybe we could create a helper function `make_account (acc : Google_Analytics_Account | Text)`?